Quadriga/src/main/java/edu/asu/spring/quadriga/utilities/impl/TextXMLParser.java

```diff
@@ -56,35 +59,46 @@ public ITextFile parseTextXML(String xml, String wsId, String projId) throws Tex
         NodeList textNodeList = document.getElementsByTagName("text");
         NodeList handleNode = document.getElementsByTagName("handle");
         NodeList fileNameNode = document.getElementsByTagName("file_name");
+        NodeList accessibilityNode = document.getElementsByTagName("accessibility");
 
         if (handleNode.getLength() == 0 && fileNameNode.getLength() == 0) {
-            throw new TextFileParseException("Handle and file name must be specified in the input XML");
+            throw new TextFileParseException("Handle and file name must be specified in the input XML.");
         } else if (handleNode.getLength() == 0) {
-            throw new TextFileParseException("Handle must be specified in the input XML");
+            throw new TextFileParseException("Handle must be specified in the input XML.");
         } else if (fileNameNode.getLength() == 0) {
-            throw new TextFileParseException("File name must be specified in the input XML");
+            throw new TextFileParseException("File name must be specified in the input XML.");
         } else if (textNodeList.getLength() == 0) {
-            throw new TextFileParseException("File content must be specified in the input XML");
+            throw new TextFileParseException("File content must be specified in the input XML.");
+        } else if (accessibilityNode.getLength() == 0) {
+            throw new TextFileParseException("Accessibility must be specified in the input XML.");
         }
 
         String fileContent = textNodeList.item(0).getTextContent();
         String fileName = fileNameNode.item(0).getTextContent();
         String refId = handleNode.item(0).getTextContent();
-        
+        String accessibility = accessibilityNode.item(0).getTextContent();
+
+        if (accessibility.isEmpty()) {
+            throw new TextFileParseException("Specify accessibility options in the XML.");
+        } else if (!(accessibility.equalsIgnoreCase("public") || accessibility.equalsIgnoreCase("private"))) {
```

Comment:
don't hardcode strings here. You can use EProjectAccessibility (but rename it to make it project independent).